### DISCUSSION TITLE
dpgen could only generate few candidates

### DISCUSSION THREAD

**Discussion by theAfish** [deepmodeling/dpgen#499] [Q&A] [answered]
Hi all, here I am again.

Recently I encountered the problem that in each iteration, there are only 0-4 candidates in model_devi task that were chosen to take the fp task, as shown in the figure below:
![49b2e6d6f9a4b1ee979cae5afd19579](https://user-images.githubusercontent.com/85424334/129662638-80a4d1e9-7d64-40fb-aa9e-81abfd77fbf6.png)

The structure I used to generate the dataset and to be explored in model_devi process is shown below:
![image](https://user-images.githubusercontent.com/85424334/129662882-bfd35391-56a6-4636-9155-e1a2088b7d7e.png)

The parameters were set with reference to the official CH4 case and some papers, using a descriptor net of [240,240,240], a fitting net of [25,50,100]. "sel" was determined with r_cut. And the detailed param.json, parameters for init_bulk (init_param.json), the model_devi.out, lcurve.out, and INCAR files for md and single-point calculations are attached below.
[param.json.txt](https://github.com/deepmodeling/dpgen/files/6997060/param.json.txt)
[init_param.json.txt](https://github.com/deepmodeling/dpgen/files/6997061/init_param.json.txt)
[model_devi.out.txt](https://github.com/deepmodeling/dpgen/files/6997063/model_devi.out.txt)
[lcurve.out.txt](https://github.com/deepmodeling/dpgen/files/6997064/lcurve.out.txt)
[INCAR_md.txt](https://github.com/deepmodeling/dpgen/files/6997065/INCAR_md.txt)
[INCAR.txt](https://github.com/deepmodeling/dpgen/files/6997066/INCAR.txt)

Can anyone help me to find out where I need to improve? Thank you!

**Comment by njzjz** [comment, marked as answer]
Your stop_batch is only 10,000, which is too small.

**Reply by theAfish** [reply]
Thank you very much! And I also found another problem: since the organic part has a large range of motion, it easily has almost all configurations failed in the model_devi parts and left only a few configurations to be explored, causing the trained model to fail to predict models in the following iterations. So I decided to set the f_trust_lo and f_trust_hi to 0.1 and 0.6 (rather than set f_trust_hi = 3 * f_trust_lo). It seems to work for this system.

**Reply by jdichongxin** [reply]
> Thank you very much! And I also found another problem: since the organic part has a large range of motion, it easily has almost all configurations failed in the model_devi parts and left only a few configurations to be explored, causing the trained model to fail to predict models in the following iterations. So I decided to set the f_trust_lo and f_trust_hi to 0.1 and 0.6 (rather than set f_trust_hi = 3 * f_trust_lo). It seems to work for this system.

Interesting discussion, my current system is melt slag, dpgen has a 99% failure rate.      I should probably increase the f_hi value in response to the rapid ion migration at high temperatures.
